corehq/apps/case_search/const.py

```diff
@@ -54,6 +54,7 @@ def es_field_name(self):
     _INDEXED_METADATA(
         key='@case_id',
         system_name='_id',
+        _es_field_name='doc_id',
```

Comment:
This is only used in a handful of places:
* sorting for the CLE
* sorting for the v0.6 API
* comparison operators in case search

The effect of this change is that sorting by `es_field_name` should work now, so I can remove the code to filter by the nested doc containing `@case_id`.  For the comparison operators, this will now query against `doc_id` instead of `_id`, which I think is an improvement.  @AmitPhulera can I get a gut check?

I'll also run a couple tests on prod to double check performance.

Reply:
Hey @esoergel!
This seems like a great optimisation. But I think we will have to wait to roll this out until we and third party hosters reindex their environments. 
As of now `doc_id` exists on all documents only on prod and other environments have it partially. 
Not sure if it would be worthwhile to add a check and make these changes specific to prod? 🤔

Reply:
Oh, does that mean other envs are still on ES 3?  I thought this was part of the ES 5 upgrade?

Reply:
No we are on ES 5 on all envs. It was part of ES 5-6 migration, we added doc_id couple of months back(I think in March)

Reply:
Ah okay - well I'll just leave this PR as a draft until that's done, then

Reply:
thanks for flagging!